builder/frameworks/ulp.py

```
@@ -31,28 +31,30 @@
     ulp_env.PrependENVPath("IDF_PATH", FRAMEWORK_DIR)
 
     toolchain_path = platform.get_package_dir(
         "toolchain-xtensa-esp-elf"
-        if idf_variant not in ("esp32c5","esp32c6", "esp32p4")
+        if is_xtensa
         else "toolchain-riscv32-esp"
     )
 
     toolchain_path_ulp = platform.get_package_dir(
         "toolchain-esp32ulp"
         if sdk_config.get("ULP_COPROC_TYPE_FSM", False)
```

Ruff (0.12.2)

Undefined name `sdk_config`

(F821)